### DISCUSSION TITLE
Physical Interfaces will be changed to virtual interfaces

### DISCUSSION THREAD

**Discussion by tkn2023** [bb-Ricardo/netbox-sync#384] [Q&A]
Hi,

I encountered an issue where the script checks the esxi host and just randomly matches existing NIC Ports and tries to change them into virtual nics which netbox does not allow because those NIC Ports are connected via cable.
Any idea what is causing the random match? No MAC Address is assigned to the NIC Port.

**Comment by bb-Ricardo** [comment]
Hi,

This is caused by the way the script marches interfaces: https://github.com/bb-Ricardo/netbox-sync/blob/main/docs/common_concepts.md

Just read this bit which describes the matching.

To solve this you would need to add the current MAC addresses to the physical ports NetBox to match your existing setup. Then the script would be able to find the correct matches.
